custom/icds/tasks.py

```diff
@@ -30,3 +50,43 @@ def run_indicator(domain, user_id, indicator_class):
 
         for message in messages:
             send_sms(domain, usercase, phone_number, message)
+
+
+def get_awc_location_ids(domain):
+    return get_location_ids_with_location_type(domain, AWC_LOCATION_TYPE_CODE)
+
+
+def get_supervisor_location_ids(domain):
+    return get_location_ids_with_location_type(domain, SUPERVISOR_LOCATION_TYPE_CODE)
+
+
+def is_first_week_of_month():
+    day = ServerTime(datetime.utcnow()).user_time(pytz.timezone('Asia/Kolkata')).done().day
+    return day <= 7
+
+
+@periodic_task(
+    run_every=crontab(hour=3, minute=30, day_of_week='mon'),
+    queue=settings.CELERY_PERIODIC_QUEUE,
+    ignore_result=True
+)
+def run_weekly_indicators():
+    """
+    Runs the weekly SMS indicators at 9am IST.
+    If it's the first week of the month, also run the monthly indicators.
+    """
+    first_week_of_month_result = is_first_week_of_month()
+
+    for domain in settings.ICDS_SMS_INDICATOR_DOMAINS:
+        for user_id in generate_user_ids_from_primary_location_ids(domain, get_awc_location_ids(domain)):
+            if first_week_of_month_result:
+                run_indicator.delay(domain, user_id, AWWAggregatePerformanceIndicator)
+
+            run_indicator.delay(domain, user_id, AWWSubmissionPerformanceIndicator)
```

Comment:
well i think we'd need to still support users who have submitted but between the 7 and 30 day window.

i was also thinking if we needed to we might be able to spread the sms load between each weekday by hashing location id or something like that and taking mod 5